File: src/main/java/com/piveguyz/empickbackend/member/query/dto/MemberResponseDTO.java
package com.piveguyz.empickbackend.member.query.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDateTime;

@Getter
@Builder
public class MemberResponseDTO {
    private Integer id;
    private Integer employeeNumber;
    private String name;
    private String email;
    private String phone;
    private String departmentName;
    private String positionName;
    private String jobName;
    private String rankName;
    private String pictureUrl;
    private Integer status;
    private LocalDateTime hireAt;
    private LocalDateTime resignAt;
}

File: src/dto/member/memberResponseDTO.js
/**
 * 회원 정보 응답 DTO
 */
export class MemberResponseDTO {
    constructor({
        id = null,
        employeeNumber = null,
        name = '',
        email = '',
        phone = '',
        departmentName = '',
        positionName = '',
        jobName = '',
        rankName = '',
        pictureUrl = '',
        status = 0,
        birth = null,
        address = '',
        hireAt = null,
        resignAt = null
    } = {}) {
        this.id = id;
        this.employeeNumber = employeeNumber;
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.departmentName = departmentName;
        this.positionName = positionName;
        this.jobName = jobName;
        this.rankName = rankName;
        this.pictureUrl = pictureUrl;
        this.status = status;
        this.birth = birth;
        this.address = address;
        this.hireAt = hireAt;
        this.resignAt = resignAt;
    }

    static fromJSON(data) {
        return new MemberResponseDTO(data);
    }
}
